components/brave_wallet_ui/components/shared/add-custom-token-form/add-custom-token-form.tsx

```diff
@@ -0,0 +1,348 @@
+        isErc721: customAssetsNetwork.coin !== BraveWallet.CoinType.SOL && !!tokenID,
+        name: tokenName,
+        symbol: tokenSymbol,
+        tokenId: tokenID ? new Amount(tokenID).toHex() : '',
+        logo: iconURL,
+        visible: true,
+        coingeckoId: coingeckoID,
+        chainId: customAssetsNetwork.chainId,
+        coin: customAssetsNetwork.coin
+      }
+      onAddCustomAsset(newToken)
+    }
+    onHideForm()
+  }, [tokenContractAddress, foundTokenInfoByContractAddress, customAssetsNetwork, iconURL, tokenDecimals, tokenName, tokenSymbol, tokenID, coingeckoID])
+
+  const onToggleShowAdvancedFields = React.useCallback(() => {
+    setShowAdvancedFields(!showAdvancedFields)
+  }, [showAdvancedFields])
+
+  const onHideNetworkDropDown = React.useCallback(() => {
+    if (showNetworkDropDown) {
+      setShowNetworkDropDown(false)
+    }
+  }, [showNetworkDropDown])
+
+  const onShowNetworkDropDown = React.useCallback(() => {
+    setShowNetworkDropDown(true)
+  }, [])
+
+  const onSelectCustomNetwork = React.useCallback((network: BraveWallet.NetworkInfo) => {
+    setCustomAssetsNetwork(network)
+    onHideNetworkDropDown()
+  }, [onHideNetworkDropDown])
+
+  const onClickCancel = React.useCallback(() => {
+    resetInputFields()
+    onHideForm()
+  }, [resetInputFields, onHideForm])
+
+  // memos
+  const isDecimalDisabled = React.useMemo((): boolean => {
+    return foundTokenInfoByContractAddress?.isErc721 ?? tokenID !== ''
+  }, [foundTokenInfoByContractAddress, tokenID])
+
+  const buttonDisabled = React.useMemo((): boolean => {
+    return tokenName === '' ||
+      tokenSymbol === '' ||
+      (tokenDecimals === '0' && tokenID === '') ||
+      tokenDecimals === '' ||
+      tokenContractAddress === '' ||
+      !customAssetsNetwork ||
+      (customAssetsNetwork?.coin !== BraveWallet.CoinType.SOL &&
+        !tokenContractAddress.toLowerCase().startsWith('0x'))
+  }, [tokenName, tokenSymbol, tokenDecimals, tokenID, tokenContractAddress, customAssetsNetwork])
+
+  // effects
+  React.useEffect(() => {
+    if (!contractAddress) return
+    setTokenContractAddress(contractAddress)
+  }, [contractAddress])
+
+  React.useEffect(() => {
+    setHasError(addUserAssetError)
+  }, [addUserAssetError])
+
+  React.useEffect(() => {
+    if (tokenContractAddress === '') {
+      resetInputFields()
+      return
+    }
+    onFindTokenInfoByContractAddress(tokenContractAddress)
+    if (foundTokenInfoByContractAddress) {
+      setTokenName(foundTokenInfoByContractAddress.name)
+      setTokenSymbol(foundTokenInfoByContractAddress.symbol)
+      setTokenDecimals(foundTokenInfoByContractAddress.decimals.toString())
+    }
+    if (foundTokenInfoByContractAddress?.isErc721) {
+      if (tokenID === '') {
+        setShowTokenIDRequired(true)
+        setShowAdvancedFields(true)
+      }
+    } else {
+      setShowTokenIDRequired(false)
+    }
+  }, [foundTokenInfoByContractAddress, tokenID, tokenContractAddress])
```

Comment:
Can this logic be moved into `handleTokenAddressChanged`?

Reply:
@josheleonard It can be moved, I just think it is better for the change handler(`handleTokenAddressChanged`) to remain in its own method.